contractsSol5/Dao/KyberDAO.sol

```diff
@@ -470,81 +439,75 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
         return totalEpochPoints[epoch] == 0;
     }
 
-    function getCampaignDetails(uint campID)
+    function getCampaignDetails(uint campaignID)
         external view
         returns(
-            CampaignType campType, uint startTimestamp, uint endTimestamp, uint totalKNCSupply,
+            CampaignType campaignType, uint startTimestamp, uint endTimestamp, uint totalKNCSupply,
             uint minPercentageInPrecision, uint cInPrecision, uint tInPrecision,
             bytes memory link, uint[] memory options
         )
     {
-        Campaign storage camp = campaignData[campID];
-        campType = camp.campType;
-        startTimestamp = camp.startTimestamp;
-        endTimestamp = camp.endTimestamp;
-        totalKNCSupply = camp.totalKNCSupply;
-        minPercentageInPrecision = camp.formulaData.minPercentageInPrecision;
-        cInPrecision = camp.formulaData.cInPrecision;
-        tInPrecision = camp.formulaData.tInPrecision;
-        link = camp.link;
-        options = camp.options;
+        Campaign storage campaign = campaignData[campaignID];
+        campaignType = campaign.campaignType;
+        startTimestamp = campaign.startTimestamp;
+        endTimestamp = campaign.endTimestamp;
+        totalKNCSupply = campaign.totalKNCSupply;
+        minPercentageInPrecision = campaign.formulaData.minPercentageInPrecision;
+        cInPrecision = campaign.formulaData.cInPrecision;
+        tInPrecision = campaign.formulaData.tInPrecision;
+        link = campaign.link;
+        options = campaign.options;
     }
 
-    function getCampaignVoteCountData(uint campID) external view returns(uint[] memory voteCounts, uint totalVoteCount) {
-        uint[] memory votes = campOptionVotes[campID];
-        if (votes.length == 0) {
-            return (voteCounts, totalVoteCount);
-        }
-        totalVoteCount = votes[0];
-        voteCounts = new uint[](votes.length - 1);
-        for (uint i = 0; i < voteCounts.length; i++) {
-            voteCounts[i] = votes[i + 1];
-        }
+    function getCampaignVoteCountData(uint campaignID) external view returns(uint[] memory voteCounts, uint totalVoteCount) {
+        CampaignVoteData memory voteData = campaignData[campaignID].campaignVoteData;
+        totalVoteCount = voteData.totalVotes;
+        voteCounts = voteData.votePerOption;
     }
 
     /**
     * @dev return campaign winning option and its value
-    * @dev return (0, 0) if campaign does not existed
-    * @dev return (0, 0) if campaign is not ended yet
-    * @dev return (0, 0) if campaign has no winning option based on the formula
-    * @param campID id of campaign to get result
+    *      return (0, 0) if campaign does not existed
+    *      return (0, 0) if campaign is not ended yet
```

Comment:
typos:
existed --> exist
is not ended --> has not ended.

Reply:
Fixed